experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/utils/EventUtils.kt

```diff
@@ -0,0 +1,26 @@
+package example.imageviewer.utils
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.input.pointer.*
+
+fun Modifier.onPointerEvent(
+    eventType: PointerEventType,
+    pass: PointerEventPass = PointerEventPass.Main,
+    onEvent: AwaitPointerEventScope.(event: PointerEvent) -> Unit
+): Modifier = composed {
+    val currentEventType by rememberUpdatedState(eventType)
+    val currentOnEvent by rememberUpdatedState(onEvent)
+    pointerInput(pass) {
+        awaitPointerEventScope {
+            while (true) {
```

Comment:
Should we use something like `isActive` to make it cancelable?

Reply:
awaitPointerEvent is suspending call, so it can be cancelled